xarray/core/computation.py

```diff
@@ -390,7 +406,9 @@ def apply_dataset_vfunc(
 
     if keep_attrs and isinstance(first_obj, Dataset):
```

Comment:
Isn't `first_obj` always a Dataset?

Reply:
yes, you are right. the dispatching takes care of this and these conditions are not necessary